src/libraries/Microsoft.Extensions.Logging.TraceSource/src/TraceSourceLogger.cs

```diff
@@ -16,7 +16,7 @@ public TraceSourceLogger(DiagnosticsTraceSource traceSource)
             _traceSource = traceSource;
         }
 
-        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string>? formatter)
```

Comment:
I think this should match the ILogger signature.

```suggestion
        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
```